src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1577,25 +1577,288 @@ class QuantizedOr(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
 
-    This operation is not really working as a quantized operation. It just works when things got
-    fused, as in e.g., Act(x) = 1000 / (x + 42))
+    Can divide either two variables (both encrypted) or a variable and a constant
     """
 
     _impl_for_op_named: str = "Div"
 
+    def __init__(
+        self,
+        *args,
+        rounding_threshold_bits: Union[None, int, Dict[str, Union[str, int]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, rounding_threshold_bits=rounding_threshold_bits, **kwargs)
+        self.divider_quantizer: UniformQuantizer
+        self.min_non_zero_value: numpy.float64
+
+    def calibrate(self, *inputs: numpy.ndarray) -> numpy.ndarray:
+        """Create corresponding QuantizedArray for the output of the activation function.
+
+        Args:
+            *inputs (numpy.ndarray): Calibration sample inputs.
+
+        Returns:
+            numpy.ndarray: the output values for the provided calibration samples.
+        """
+
+        # If two inputs, compute the divider
+        if not self.can_fuse() and len(inputs) == 2:
+            min_non_zero_value = numpy.min(numpy.abs(inputs[1]))
+
+            # mypy
+            assert min_non_zero_value is not None and min_non_zero_value > 0
+            self.min_non_zero_value = min_non_zero_value
+
+            q_array_divider = QuantizedArray(self.n_bits, 1 / inputs[1])
+
+            # Store the quantizer of the divider
+            self.divider_quantizer = q_array_divider.quantizer
+
+        return super().calibrate(*inputs)
+
+    def q_impl(
+        self,
+        *q_inputs: ONNXOpInputOutputType,
+        calibrate_rounding: bool = False,
+        **attrs,
+    ) -> ONNXOpInputOutputType:
+
+        # If the op can be fused or if any input
+        # is a RawOpOutput, perform the op in the clear
+        if (
+            len(q_inputs) == 1
+            or any(isinstance(input, RawOpOutput) for input in q_inputs)
+            or self.can_fuse()
+        ):
+            return super().q_impl(*q_inputs, **attrs)
+
+        # For mypy
+        assert self.output_quant_params is not None
+        assert self.output_quant_params.scale is not None
+        assert self.output_quant_params.zero_point is not None
+
+        prepared_inputs = self._prepare_inputs_with_constants(
+            *q_inputs, calibrate=False, quantize_actual_values=True
+        )
+
+        q_input_0: QuantizedArray = prepared_inputs[0]
+        q_input_1: QuantizedArray = prepared_inputs[1]
+
+        assert q_input_0.quantizer.scale is not None
+        assert q_input_0.quantizer.zero_point is not None
+
+        assert q_input_1.quantizer.scale is not None
+        assert q_input_1.quantizer.zero_point is not None
+
+        # Dequantize
+        input_1 = q_input_1.dequant()
+
+        # Replace input_1 with min_non_zero_qvalue if input_1 is 0
+        input_1 = numpy.where(input_1 == 0, self.min_non_zero_value, input_1)
```

Comment:
how is this working ? just replacing 0 with the (float) min ? I imagine that means that values are expected to be positive ! And then still, should we instead add `self.min_non_zero_value` to ìnput_1`, instead of replacing 0, right ? 

in any case we should explain in comments how all this work

Reply:
so what should we do here then ?

Reply:
Actually, here we are within a PBS and we assign values that are 0 to self.min_non_zero_value. That should work fine within a PBS.

Reply:
yes but my observation was that, since we replace all zeros with `self.min_non_zero_value` : 
- since this value is computed through `numpy.min(numpy.abs(inputs[1]))` , does that mean we expect `input_1` to always be positive ? 
    - if not, for example, `[-2, 4, 0]` would be replaced by `[-2, 4, 2]` and I don't think it makes much sense
    - if so, how can we be sure that `input_1` is `>= 0` ? more that having asserts I mean (like my comment in https://github.com/zama-ai/concrete-ml/pull/690/files#r1677747936)
- also, aren't we breaking the values' distribution by doing something like this ? initially I suggested to instead do something like `input_1 += min_non_zero_value` (or `input_1 += ` if positive) but don't think that's desirable as well. What about requantizing on [1, max_val] ? Anyway, I feel just replacing 0 by this new val is not exactly right

Reply:
Ah yes you are right good catch! I should keep the sign of the value. I fixed it.

Reply:
Done

Reply:
hmm well although that fixes the issue, I don't think it's right though now ? eg if we have a `[-20, 3, 0, 4]`, your code changes it to `[-20, 3, -20, 4]`, which drastically changes the values' distribution ⚠️  This is basically what I was trying to say in my second point : 
> also, aren't we breaking the values' distribution by doing something like this ? initially I suggested to instead do something like input_1 += min_non_zero_value (or input_1 +=  if positive) but don't think that's desirable as well. What about requantizing on [1, max_val] ? Anyway, I feel just replacing 0 by this new val is not exactly right

imo what we are doing here is still wrong !

Reply:
> hmm well although that fixes the issue, I don't think it's right though ? eg if we have a [-20, 3, 0, 4], your code changes it to [-20, 3, -20, 4], which drastically changes the values' distribution ⚠️ This is basically what I was trying to say in my second point 

This input_1 are floating points. Here it would change `[-20, 3, 0, 4]` to `[-20, 3, 3, 4]`. Why ``[-20, 3, 3, 4]`. We don't take the min we take min of abs.

Another way we could do this is to take the smallest floating point represented in the quantized version of inputs[1]. But we don't have the quant information for this so I am not sure this is robust.

We can discuss this offline if needed.

Reply:
I totally understand your worry on this however! Let's find the best solution.

Reply:
oh ok sorry I don't know why I kept seeing `argmax` and not `argmin`, so yeah ok we just get the closest value to 0, makes much more sense indeed

Reply:
would it make sense to instead add an epsilon to this 0 ? or simply to all values ? not sure it'll change much since we re-quantize after the 1/x

apart from that, I'm not sure I have better ideas here

Reply:
> would it make sense to instead add an epsilon to this 0 ?

Yes that's what I proposed. An epsilon would make sense if it's represented by the quantized values. The scale is basically the smallest represented floating point. So we could use this instead of the min(abs(x)). But I am not entirely sure we can get the actual quantizer of the input. Also I am afraid that for large bit-width this epsilon is going to be super small and will lead to numerical errors.

Reply:
yeah numerical errors could be an issue indeed. But actually this epsilon should be added to all values not just the 0 maybe ? in order to keep the same distribution ? not sure how the quant params will handle this though